internal/caddy/convert.go

```diff
@@ -25,18 +25,32 @@ func ConvertToCaddyConfig(ings []*v1beta1.Ingress) (caddyhttp.RouteList, error)
 				clusterHostName := fmt.Sprintf("%v.%v.svc.cluster.local:%d", path.Backend.ServiceName, ing.Namespace, path.Backend.ServicePort.IntVal)
 				r := baseRoute(clusterHostName)
 
-				r.MatcherSetsRaw = []caddy.ModuleMap{
-					{
-						"host": json.RawMessage(`["` + fmt.Sprintf("%s", rule.Host) + `"]`),
-						"path": json.RawMessage(`["` + fmt.Sprintf("%s", path.Path) + `"]`),
-					},
+				match := caddy.ModuleMap{}
+
+				if rule.Host != "" {
+					hostRule, err := json.Marshal(caddyhttp.MatchHost{rule.Host})
```

Comment:
Definitely use [caddyconfig.JSON](https://pkg.go.dev/github.com/caddyserver/caddy/v2/caddyconfig?tab=doc#JSON) instead. (or [JSONModuleObject](https://pkg.go.dev/github.com/caddyserver/caddy/v2/caddyconfig?tab=doc#JSONModuleObject) if the JSON object has to add the module name)